handler.fish

```diff
@@ -0,0 +1,21 @@
+function __fish_command_not_found_on_interactive --on-event fish_prompt
+    functions --erase __fish_command_not_found_handler
```

Comment:
This is used to erase the default `__fish_command_not_found_handler` so we won't have duplicated `command not found` message.

> Some people have custom command not found handler function

Can you show me an example about that? If they need a hook, shouldn't them just use `--on-event fish_command_not_found`? I don't think people want to have multiple command not found functions at the same time.